plugin-guide.md

```diff
@@ -315,3 +368,370 @@ provided in the metadata.
+}
+```
+
+The input will take a tagged union/algebraic data type with the appropriate
+inputs. If the name provided by the submitter does not match one of the inputs,
+`"defaultPriority"` is used instead. The names of the keys of `"inputs"` should
+be capitalized for compatibility with Shesmu.
+
+### Prometheus Input
+Reads a variable from Prometheus, filtering on the label set, and returns the
+current value.
+
+```
+{
+  "type": "prometheus",
+  "cacheRequestTimeout": 1,
+  "cacheTtl": 15,
+  "defaultPriority": 0,
+  "labels": ["bob"],
+  "query": "some_prometheus_variable",
+  "url": "http://prometheus.example.com:9090",
+  "workflowNameLabel": "workflow",
+  "workflowVersionLabel": null
+}
+```
+
+The process this input provider uses is as follows:
+
+1. Execute `"query"` on the Prometheus instance at `"url"`. The query can be
+   any valid Prometheus query. If it takes longer than `"cacheRequestTimeout"`
+   minutes, then the query will be treated as a failure. The results will be
+   cached for `"cacheTtl"` minutes before being refreshed.
+2. The submission request will be processed into a label set as described below.
+3. All the records that were returned by the query are scanned for a matching
+   label set.
+4. If a matching label set is found, the last recorded value will be used,
+   regardless of when Prometheus observed it.
+5. If no matching label set is found, `"defaultPriority"` will be  used.
+
+The label set is constructed from the submission request. For each string in
+`"labels"`, the submitter must provide a string value. These labels and values
+will be used as the label set. For example, with the configuration `"labels":
+["bob"]`, the submission request could have `{"bob": "eggs"}` and the filtered
+label set would look like `[bob=eggs]`. Additionally, special labels are
+available for the workflow name and version. If `"workflowNameLabel":
+"workflow"` and the submission request was for `bcl2fastq`, then the label set
+would be `[workflow=bcl2fastq]`. This can be further refined with a workflow
+version using `"workflowVersionLabel"`, which will only be used if
+`"workflowNameLabel"` is not null. Both of these can be turned off by being set
+to null.
+
+### Raw Priority Input
+Takes an optional integer from the submission request and returns it raw, or
+`"defaultPriority"` if not provided.
+
+```
+{
+  "type": "raw",
+  "defaultPriority": 0
+}
+```
+
+### Remote Input
+Takes an arbitrary JSON value and sends it to remote HTTP endpoint for
+evaluation. That endpoint must return a single number. The result will be
+cached. The `"schema"` is a stanadard Víðarr type that should be requested from
+the submission request.
+
+```
+{
+  "type": "remote",
+  "defaultPriority": 0,
+  "schema": "string",
+  "ttl": 15,
+  "url": "http://foo.com/api/get-priority"
+}
+```
+
+The `"schema"` property defines a type, including an object types, that will be
+required on submission. The data provided by the submission will be send via
```

Comment:
will be send -> sent

Reply:
I'm not sure what's outstanding here. There's only one _Remote Input_ now.